proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -1015,22 +1017,36 @@ private void tickResourcePackQueue() {
         request.setHash("");
       }
       request.setRequired(queued.getShouldForce());
-      request.setPrompt(queued.getPrompt());
+      request.setPrompt(queued.getPrompt() == null ? null : new ComponentHolder(getProtocolVersion(), queued.getPrompt()));
 
       connection.write(request);
     }
   }
 
   @Override
+  @Deprecated
   public @Nullable ResourcePackInfo getAppliedResourcePack() {
+    //TODO which resource pack should be returned here?
     return appliedResourcePack;
   }
 
   @Override
+  @Deprecated
   public @Nullable ResourcePackInfo getPendingResourcePack() {
+    //TODO which resource pack should be returned here?
     return pendingResourcePack;
   }
 
+  @Override
+  public Collection<ResourcePackInfo> getAppliedResourcePacks() {
+    //TODO
```

Comment:
Shouln't this be finished before a merge?

Reply:
Yes, that's the entire point it was converted into a PR.